packages/vitest/src/node/core.ts

```diff
@@ -379,6 +383,79 @@ export class Vitest {
     return Promise.all(this.projects.map(w => w.initBrowserProvider()))
   }
 
+  async mergeReports() {
+    const blobs = await readBlobs(this.config.mergeReports)
+
+    if (!blobs.length)
+      throw new Error(`vitest.mergeReports() requires at least one blob file paths in the config`)
+
+    await this.report('onInit', this)
+
+    // fake module graph - it is used to check if module is imported, but we don't use values inside
+    const projects = Object.fromEntries(this.projects.map(p => [p.getName(), p]))
+
+    blobs.forEach((blob) => {
+      blob.moduleKeys.forEach(([projectName, moduleIds]) => {
+        const project = projects[projectName]
+        if (!project)
+          return
+        moduleIds.forEach((moduleId) => {
+          project.server.moduleGraph.idToModuleMap.set(moduleId, {
+            id: moduleId,
+            url: moduleId,
+            file: moduleId,
+            ssrTransformResult: null,
+            transformResult: null,
+            importedBindings: null,
+            importedModules: new Set(),
+            importers: new Set(),
+            type: 'js',
+            clientImportedModules: new Set(),
+            ssrError: null,
+            ssrImportedModules: new Set(),
+            ssrModule: null,
+            acceptedHmrDeps: new Set(),
+            acceptedHmrExports: null,
+            lastHMRTimestamp: 0,
+            lastInvalidationTimestamp: 0,
+          })
+        })
+      })
+    })
+
+    const files = blobs.flatMap(blob => blob.files).sort((f1, f2) => {
+      const time1 = f1.result?.startTime || 0
+      const time2 = f2.result?.startTime || 0
+      return time1 - time2
+    })
+    const errors = blobs.flatMap(blob => blob.errors)
+    this.state.collectFiles(files)
+
+    await this.report('onCollected', files.map((file) => {
+      return { ...file, result: undefined }
+    }))
+
+    for (const file of files) {
+      const logs: UserConsoleLog[] = []
+      const taskPacks: TaskResultPack[] = []
+
+      const tasks = getTasks(file)
+      for (const task of tasks) {
+        if (task.logs)
+          logs.push(...task.logs)
+        taskPacks.push([task.id, task.result, task.meta])
+      }
+      logs.sort((log1, log2) => log1.time - log2.time)
```

Comment:
Check if it makes sense to instead move some of this logic into `./reporters/blob` directly. Would be nice if `core.ts` would only call `this.report(...)` with the results returned from the reporter. And to avoid growing `core.ts` bigger than it already is.

Reply:
I don't really understand what your proposal is. The whole idea is to report via other reporters, not _blob_ reporter, the blob reporter is ignored (probably will throw an error) if `--merge-reports` is used since we already generated a report

Reply:
Something like this maybe. Just to move some blob/file preparation logic to reporter itself. Maybe the task and log part could also be moved there.

<details>

```ts
  async mergeReports() {
    const projects = Object.fromEntries(this.projects.map(p => [p.getName(), p]))
    const { files, errors } = await readBlobs(this.config.mergeReports, projects)

    if (!files.length)
      throw new Error(`vitest.mergeReports() requires at least one blob file paths in the config`)

    await this.report('onInit', this)

    this.state.collectFiles(files)
    await this.report('onCollected', files.map(file => ({ ...file, result: undefined })))

    for (const file of files) {
      const logs: UserConsoleLog[] = []
      const taskPacks: TaskResultPack[] = []

      const tasks = getTasks(file)
      for (const task of tasks) {
        if (task.logs)
          logs.push(...task.logs)
        taskPacks.push([task.id, task.result, task.meta])
      }
      logs.sort((log1, log2) => log1.time - log2.time)

      for (const log of logs)
        await this.report('onUserConsoleLog', log)

      await this.report('onTaskUpdate', taskPacks)
    }

    await this.report('onFinished', files, errors)
  }
```

```ts

export async function readBlobs(blobsDirectory: string, projects: Record<string, WorkspaceProject>) {
  const resolvedDir = resolve(process.cwd(), blobsDirectory)
  const filenames = await readdir(resolvedDir)

  const blobs = await Promise.all(filenames.map(async (file) => {
    const content = await readFile(resolve(resolvedDir, file), 'utf-8')
    const [version, files, errors, moduleKeys] = parse(content) as [string, files: File[], errors: unknown[], [string, string[]][]]
    return { version, files, errors, moduleKeys }
  }))

  // fake module graph - it is used to check if module is imported, but we don't use values inside
  blobs.forEach((blob) => {
    blob.moduleKeys.forEach(([projectName, moduleIds]) => {
      const project = projects[projectName]
      if (!project)
        return
      moduleIds.forEach((moduleId) => {
        project.server.moduleGraph.idToModuleMap.set(moduleId, {
          id: moduleId,
          url: moduleId,
          file: moduleId,
          ssrTransformResult: null,
          transformResult: null,
          importedBindings: null,
          importedModules: new Set(),
          importers: new Set(),
          type: 'js',
          clientImportedModules: new Set(),
          ssrError: null,
          ssrImportedModules: new Set(),
          ssrModule: null,
          acceptedHmrDeps: new Set(),
          acceptedHmrExports: null,
          lastHMRTimestamp: 0,
          lastInvalidationTimestamp: 0,
        })
      })
    })
  })

  const errors = blobs.flatMap(blob => blob.errors)
  const files = blobs.flatMap(blob => blob.files).sort((f1, f2) => {
    const time1 = f1.result?.startTime || 0
    const time2 = f2.result?.startTime || 0
    return time1 - time2
  })

  return { files, errors }
}
```

</details>